### ISSUE TITLE
chatapp doesnot work anymore

### ISSUE THREAD

**Issue by stickylabdev** [Dual2Core/Smile#1]
Please fix .why i cant chat with myfrient?

**Comment by pkobielak** [comment]
Did you insert your Parse API key and credentials into the code? If you do, then there shouldn't be any problems. Please elaborate your question.

**Comment by stickylabdev** [comment]
Surely i  change api key and client id with my credential in parse.com

**Comment by stickylabdev** [comment]
Surei cannot see any chat come in ...i see just my chat

**Comment by pkobielak** [comment]
Ofcourse it is like that. Because you have to change every string "Girlfriend" for "Boyfriend" on the second client to make it work. Or use my desktop application which i was using as default second client. I will upload its project in a moment.

EDIT: It is published now. Check my repository Smile-Desktop. Turn them on simulatenously and it will work for sure. Waiting for your feedback.

**Comment by stickylabdev** [comment]
Desktop PC? But i have not computer yet ..better if you make login credential like this project made by me goo.gl/dbnTwL

**Comment by stickylabdev** [comment]
Oh .i will  install in one android phone .and make 2 app ..one as girl and other as boy

**Comment by stickylabdev** [comment]
Sorry for bad English .

**Comment by stickylabdev** [comment]
So ijust change value in the 
     ChatMessage.java 

```
 @Override
 public void parse(ParsingComponent cloud)
 {
      ParseObject msg = new ParseObject("Messages");
      msg.put("text", message);
      msg.put("sender", "BoyFriend");
      msg.put("readstate", false);
      msg.pinInBackground();
      msg.saveInBackground();
 }
```

**Comment by stickylabdev** [comment]
Sadly ..i your app only work on local android phone  ...when i try with onther phone ..it doesnot work

**Comment by stickylabdev** [comment]
WOW THIS IS REALLY REALY WORK .YEAH.

**Comment by stickylabdev** [comment]
But i need you to add sign up and login system ..i am little confuse if i do it myself

**Comment by pkobielak** [comment]
You can do it bro if you want. I'll merge your solution if it would work. I have no time for it currently.

**Comment by stickylabdev** [comment]
Yes bro .but ifound bug on your code brow . If it no problem when i open the app for first time .but .when i close then open again ,it force closed :'(

**Comment by pkobielak** [comment]
If you dont have internet connection during opening the application or you lose it during its activity, then the app will crash. Again, I am working on different project now. :)

**Comment by stickylabdev** [comment]
Bro .i realy sure i have internet access ..i think the problem is Thread .

**Comment by stickylabdev** [comment]
The cronologis .is like this.
1.i open the app for fist time .realy work .i can chat and anything! Then exit .
2.i open again .then force close .force me to close .then
3.i open again.them perfectly works !
4.:( .
How to fix ;(

**Comment by pkobielak** [comment]
Exit detection should be implemented, you can do it If you want. As i said the project is not developed by me now. The issue will be closed.

**Comment by stickylabdev** [comment]
Exit detection should be implemented, 

So ijust add onbackpressed finish

**Comment by pkobielak** [comment]
That should do the job.

**Comment by stickylabdev** [comment]
But some tima .the messages repeat several time :"(  though i enter one time ..it always appear several messages in other app
